tasks/push-rpm-manifests-to-pyxis/push-rpm-manifests-to-pyxis.yaml

```diff
@@ -50,20 +50,36 @@ spec:
             exit 1
         fi
 
-        IMAGEURLS=($(cat "${PYXIS_FILE}" | jq -r '.components[].pyxisImages[].containerImage' | sort | uniq))
+        NUM_COMPONENTS=$(jq '.components | length' "${PYXIS_FILE}")
 
         mkdir /workdir/sboms
         cd /workdir/sboms
 
-        for i in ${!IMAGEURLS[@]}; do
-          echo "Fetching sbom for image: ${IMAGEURLS[$i]}"
-          FILE="$(echo ${IMAGEURLS[$i]} | tr "/@:" ---).json"
-          cosign download sbom --output-file "${FILE}" "${IMAGEURLS[$i]}"
+        for (( i=0; i < $NUM_COMPONENTS; i++ )); do
+          COMPONENT=$(jq -c --argjson i "$i" '.components[$i]' "${PYXIS_FILE}")
+          IMAGEURL=$(jq -r '.containerImage' <<< "${COMPONENT}")
+          NUM_PYXIS_IMAGES=$(jq '.pyxisImages | length' <<< "${COMPONENT}")
+          for (( j=0; j < $NUM_PYXIS_IMAGES; j++ )); do
+            PYXIS_IMAGE=$(jq -c --argjson j "$j" '.pyxisImages[$j]' <<< "${COMPONENT}")
+            FILE="$(jq -r '.imageId' <<< "$PYXIS_IMAGE").json"
+            # You can't pass --platform to a single arch image or cosign errors
+            if [ "${NUM_PYXIS_IMAGES}" -eq 1 ] ; then
```

Comment:
I have just one concern here. It shouldn't matter if it's one or more archs, what matters is if it's a manifest list or the other type. Some time ago I heard someone saying that they're gonna switch to manifest list style images even for single arch builds. But maybe it's not actually a problem? Because if it's a manifest list / multiarch image, but with just one arch, you can still run `cosign sbom download` without specifying `--platform` and it will still fetch the one? WDYT?

Reply:
I don't think I am fully following. How is it a multiarch image if it just has one arch? My goal is to avoid rerunning the `get-image-architectures` tool and just consume pyxis.json. So maybe you are saying store the manifest list or other type in that and use that in the if?

Reply:
It's the reason we have [this](https://github.com/konflux-ci/release-service-utils/blob/main/pyxis/create_container_image.py#L51):
```
# Media types that are used for multi arch images
MANIFEST_LIST_TYPES = [
    "application/vnd.oci.image.index.v1+json",
    "application/vnd.docker.distribution.manifest.list.v2+json",
]
```
You can have an image, which only has one arch, but it's inside a manifest list - so the image has `"media_type": "application/vnd.docker.distribution.manifest.list.v2+json",`. That's what I mean. So for all purposes it's a multiarch image.

So for such an image, you will run `cosign sbom download` without the `--platform` argument. Will it still work? Probably.

Reply:
Yeah, I don't know of such image either. As I said, maybe the build service will switch to these images in the future. I guess we'll have to wait and see...

Reply:
Yeah, good to think of for if/when it happens